### DISCUSSION TITLE
Tasmota remote control

### DISCUSSION THREAD

**Discussion by lionlggnd** [arendst/Tasmota#17947] [Q&A]
i need to control my tasmota-programmed device away my home, too.
How i can do this?
On internet i found that i need MQTT.
I've searched a lot about this, but online guides are scarce and I haven't been able to understand how it works well.
Can i have a detailed explaination of MQTT and the parameters i have to configure in tasmota and their functions?
Can i have, also, reccomendations about some free and reliable MQTT service can i control with a IOS/ANDROID app?
Thanks.

**Comment by barbudor** [comment]
I'm using flespi.io. Free until ~20 devices (above there's only professionnal plans unsuitable for hobbyist)
I've also tested MyMqttHub. It worked but I choose flespi for some reason.
Both need encrypted MQTT (TLS) which is not build by default in Tasmota for ESP8266 but is in tasmota32 for esp32

On Android I'm using "IoT MQTT Panel", free version

Dealing with MQTT over Internet may not be the simplest to start with for beginners if you don't yet have some understanding of mqtt

**Reply by princekham** [reply]
How did you connect to flespi.io with esp32 Tasmota? I am trying to do it and not alright yet. Thanks!

**Reply by barbudor** [reply]
For connecting to flespi here are the parameters that I'm using to build my custom binaries
Most are already available on standard ESP32 binary and others can be entered on the console
But that's a ood summary of what is needed

```
    #ifdef MQTT_HOST
      #undef MQTT_HOST
    #endif
    #define MQTT_HOST         "mqtt.flespi.io" // [MqttHost]

    #ifdef MQTT_PORT
      #undef MQTT_PORT
    #endif
    #define MQTT_PORT         8883              // [MqttPort] MQTT port

    #ifndef USE_MQTT_TLS
      #define USE_MQTT_TLS                             // Use TLS for MQTT connection (+34.5k code, +7.0k mem and +4.8k additional during connection handshake)
    #endif
    #ifdef MQTT_TLS_ENABLED
      #undef MQTT_TLS_ENABLED
    #endif
    #define MQTT_TLS_ENABLED  true             // [SetOption103] Enable TLS mode (requires TLS version)
    #ifdef MQTT_TLS_FINGERPRINT
      #undef MQTT_TLS_FINGERPRINT
    #endif
    #define MQTT_TLS_FINGERPRINT   true             // [SetOption132] Force TLS fingerprint validation instead of CA (requires TLS version)

    #ifdef MQTT_USER
      #undef MQTT_USER
    #endif
    #define MQTT_USER         "xxxxxxxxxxxxxxxxxxxxxx" // Enter device token as created in the Flespi dashboard

    #ifdef MQTT_PASS
      #undef MQTT_PASS
    #endif
    #define MQTT_PASS         ""       // [MqttPassword] Optional password
```

When creating the token, you need to have the MQTT pub+sub options
![image](https://github.com/user-attachments/assets/919e3e0a-0dc0-4053-b5ab-152e20c88a68)

**Comment by lionlggnd** [comment]
Can you explain me the mqtt?

**Reply by barbudor** [reply]
No
There is a brief introduction in Tasmota docs with links to tutorials in the 1st paragraph

https://tasmota.github.io/docs/MQTT/
